include/sound/soc_sdw_utils.h

```diff
@@ -28,6 +28,7 @@
  *   - SOC_SDW_CODEC_SPKR | SOF_SIDECAR_AMPS - Not currently supported
  */
 #define SOC_SDW_SIDECAR_AMPS		BIT(16)
+#define SOC_SDW_CODEC_NO_MIC		BIT(17)
```

Comment:
Maybe just `SOC_SDW_CODEC_MIC` without `NO`? We have `quirk_exclude` to represent default value with or without the device. So we don't need to represent it in the quirk name, and save some quirk bits. I.e. Other codecs can use the `SOC_SDW_CODEC_MIC` quirk to don't use codec mic by default.

Reply:
Also, can we split it to 2 commits? One to add the `quirk_exclude` flag and the other to make the cs42l43 MIC optional.

Reply:
Yeah that makes sense will do that.